src/Altinn.Platform/Altinn.Platform.Events/Functions/Services/KeyVaultService.cs

```diff
@@ -15,12 +17,23 @@ namespace Altinn.Platform.Events.Functions.Services
     public class KeyVaultService : IKeyVaultService
     {
         /// <inheritdoc/>
-        public async Task<string> GetSecretAsync(string vaultUri, string secretId)
+        public async Task<string> GetCertificateAsync(string vaultUri, string secretId)
         {
-            SecretClient secretClient = new SecretClient(new Uri(vaultUri), new DefaultAzureCredential());
-            KeyVaultSecret secret = await secretClient.GetSecretAsync(secretId);
+            CertificateClient certificateClient = new CertificateClient(new Uri(vaultUri), new DefaultAzureCredential());
```

Comment:
have you considered reusing the client? As we are only connecting to one KV, we don't need to create the client each time.